messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ConnectorAdd.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.messaging.activemq;
+
+import static org.wildfly.extension.messaging.activemq.ActiveMQReloadRequiredHandlers.isServiceInstalled;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.capability.CapabilityServiceSupport;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2023 Red Hat, Inc.
+ */
+public class ConnectorAdd extends AbstractAddStepHandler {
+
+
+    public ConnectorAdd(AttributeDefinition... attributes) {
+        super(attributes);
+    }
+
+    @Override
+    protected boolean requiresRuntime(OperationContext context) {
+        return context.isDefaultRequiresRuntime() && !context.isBooting();
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+        if (MessagingServices.isSubsystemResource(context)) {
+            return;
+        }
+        if (isServiceInstalled(context)) {
```

Comment:
What about the case where we are not booting and the service isn't installed?

Reply:
maybe I can move it to requiresRuntime too